In [3]:
from promptsmith.dspy_init import get_dspy
dspy, lm = get_dspy()

# read text_01.txt

In [13]:
with open("../data/text_01.txt", "r") as f:
    text_to_bulletize = f.read()

In [14]:
print(text_to_bulletize)

The first question in this study stated: What is the strength and direction of the relationship between faculty's perceptions of the importance of course design, communication, time management, and technical competency, and their ability to teach online? To address this question, descriptive statistics (mean and standard deviation) by individual competency items of each of the four competency dimensions: course design, course communication, time management, and technical competency,  were reported in Table 2 on page 23. Most of the items in each competency were rated high for both constructs, Importance, and Ability.
The items manage grades online (M = 4.73) and creating online assignments (M = 4.68)

were rated the highest in the course design competency. In the course communication competency, responding to student questions promptly (M = 4.79) and providing feedback on assignments (M = 4.65) were rated the highest. In time management, scheduling time to design the course prior to de

# `text_to_bulletize`

The first question in this study stated: What is the strength and direction of the relationship between faculty's perceptions of the importance of course design, communication, time management, and technical competency, and their ability to teach online? To address this question, descriptive statistics (mean and standard deviation) by individual competency items of each of the four competency dimensions: course design, course communication, time management, and technical competency,  were reported in Table 2 on page 23. Most of the items in each competency were rated high for both constructs, Importance, and Ability.
The items manage grades online (M = 4.73) and creating online assignments (M = 4.68)

were rated the highest in the course design competency. In the course communication competency, responding to student questions promptly (M = 4.79) and providing feedback on assignments (M = 4.65) were rated the highest. In time management, scheduling time to design the course prior to delivery (M = 4.65) and spending weekly hours to grade assignments (M = 42 4.54) were rated the highest. In the technical competency, complete basic computer operations (M = 4.74) and navigating within the course in the learning management system (M = 4.67) were rated the highest. As the findings show that competencies in importance for course design and course communication were scored the highest, more support is likely needed for faculty in such competencies.
In the course design competency, manage grades online (M = 4.61) and create online assignments (M = 4.60) were rated the highest items. In course communication, sending announcements/email reminders to course participants (M = 4.82), and use email to communicate with the learners (M = 4.79) were rated the highest items. In time management, schedule weekly hours to facilitate the online course (M = 4.39) and spend weekly hours to grade assignments (M = 4.37) were rated the highest. The complete basic computer operations item (M = 4.64) and navigate within the course in the learning management system item (M = 4.36) were rated the highest in the technical competency. Participants in this study perceived that the top competencies of perception of Importance are similar to the top competencies that faculty perceive as they were able to perform. This Dimension and one of the top competencies in the Time Management Dimension.

In testing the statistical significance of the relationship between self-reported perceptions of the Importance of online teaching competencies and self-reported Ability to teach online, there was a weak positive relationship between the two constructs, r ( 223) = .249, p = <.001.
The second question of this study stated: What demographic factors are related to faculty's perceptions of the importance of course design, communication, time management, and technical competency, and their ability to teach online? In order to address this research question, the demographic information collected from the survey was analyzed. Demographics included participant gender, academic rank, online teaching methods, and level taught. The analysis of the descriptive statistics (mean and standard deviation) of demographic data were presented in Table 5 of page 32, in Chapter IV. Results from ANOVA showed a significant difference in gender and level taught on importance of online teaching competencies. Results from ANOVA also showed no significant differences in gender, academic rank, and online teaching methods, but showed a significant difference in the level taught based on their perceptions of ability to teach online. No significant differences were found in other demographic factors, such as academic rank and online teaching methods. Below is a detailed analysis of the demographic information collected.

# using a simple signature: "text -> bulletized_text"

In [15]:
bulletize_text_1 = dspy.Predict("text -> bulletized_text")
result_1 = bulletize_text_1(text=text_to_bulletize)
bulletized_text_1 = result_1.bulletized_text
print(bulletized_text_1)

- The first question addressed the relationship between faculty's perceptions of course design, communication, time management, and technical competency, and their ability to teach online.
- Descriptive statistics (mean and standard deviation) for each competency dimension were reported in Table 2 on page 23.
- Most items in each competency were rated high for both constructs: Importance and Ability.
- Highest ratings in course design competency:
  - Manage grades online (M = 4.73)
  - Creating online assignments (M = 4.68)
- Highest ratings in course communication competency:
  - Responding to student questions promptly (M = 4.79)
  - Providing feedback on assignments (M = 4.65)
- Highest ratings in time management:
  - Scheduling time to design the course prior to delivery (M = 4.65)
  - Spending weekly hours to grade assignments (M = 4.54)
- Highest ratings in technical competency:
  - Complete basic computer operations (M = 4.74)
  - Navigating within the course in the learning man

# `bulletized_text_1`

- The first question addressed the relationship between faculty's perceptions of course design, communication, time management, and technical competency, and their ability to teach online.
- Descriptive statistics (mean and standard deviation) for each competency dimension were reported in Table 2 on page 23.
- Most items in each competency were rated high for both constructs: Importance and Ability.
- Highest ratings in course design competency:
  - Manage grades online (M = 4.73)
  - Creating online assignments (M = 4.68)
- Highest ratings in course communication competency:
  - Responding to student questions promptly (M = 4.79)
  - Providing feedback on assignments (M = 4.65)
- Highest ratings in time management:
  - Scheduling time to design the course prior to delivery (M = 4.65)
  - Spending weekly hours to grade assignments (M = 4.54)
- Highest ratings in technical competency:
  - Complete basic computer operations (M = 4.74)
  - Navigating within the course in the learning management system (M = 4.67)
- Findings indicate that competencies in importance for course design and communication were scored the highest, suggesting a need for more faculty support in these areas.
- Additional ratings in course design competency:
  - Manage grades online (M = 4.61)
  - Create online assignments (M = 4.60)
- Additional ratings in course communication:
  - Sending announcements/email reminders (M = 4.82)
  - Using email to communicate with learners (M = 4.79)
- Additional ratings in time management:
  - Schedule weekly hours to facilitate the online course (M = 4.39)
  - Spend weekly hours to grade assignments (M = 4.37)
- Additional ratings in technical competency:
  - Complete basic computer operations (M = 4.64)
  - Navigate within the course in the learning management system (M = 4.36)
- Participants perceived that top competencies of Importance align with those they feel able to perform.
- Statistical testing revealed a weak positive relationship between self-reported perceptions of Importance and Ability to teach online (r(223) = .249, p < .001).
- The second question examined demographic factors related to faculty's perceptions of competencies.
- Demographic data included gender, academic rank, online teaching methods, and level taught.
- Descriptive statistics for demographic data were presented in Table 5 on page 32, Chapter IV.
- ANOVA results showed significant differences in gender and level taught regarding the importance of online teaching competencies.
- No significant differences were found in gender, academic rank, and online teaching methods, but significant differences were noted in level taught based on perceptions of ability to teach online.
- No significant differences were found in other demographic factors such as academic rank and online teaching methods.

# the same simple signature, but now using 'chain of thought'

In [ ]:
bulletize_text_2 = dspy.ChainOfThought("text -> bulletized_text")
result_2 = bulletize_text_2(text=text_to_bulletize)
bulletized_text_2 = result_2.bulletized_text
print(bulletized_text_2)

- **Study Focus**: Examines the relationship between faculty perceptions of online teaching competencies and their ability to teach online.
- **Competency Dimensions Analyzed**:
  - **Course Design**: 
    - Highest rated items: 
      - Manage grades online (M = 4.73)
      - Create online assignments (M = 4.68)
  - **Course Communication**: 
    - Highest rated items: 
      - Responding to student questions promptly (M = 4.79)
      - Providing feedback on assignments (M = 4.65)
  - **Time Management**: 
    - Highest rated items: 
      - Scheduling time to design the course (M = 4.65)
      - Spending weekly hours to grade assignments (M = 4.54)
  - **Technical Competency**: 
    - Highest rated items: 
      - Complete basic computer operations (M = 4.74)
      - Navigating within the learning management system (M = 4.67)
- **Findings**: 
  - High ratings for importance and ability in course design and communication suggest a need for more faculty support in these areas.
  - Weak

# `bulletized_text_2`

- **Study Focus**: Examines the relationship between faculty perceptions of online teaching competencies and their ability to teach online.
- **Competency Dimensions Analyzed**:
  - **Course Design**: 
    - Highest rated items: 
      - Manage grades online (M = 4.73)
      - Create online assignments (M = 4.68)
  - **Course Communication**: 
    - Highest rated items: 
      - Responding to student questions promptly (M = 4.79)
      - Providing feedback on assignments (M = 4.65)
  - **Time Management**: 
    - Highest rated items: 
      - Scheduling time to design the course (M = 4.65)
      - Spending weekly hours to grade assignments (M = 4.54)
  - **Technical Competency**: 
    - Highest rated items: 
      - Complete basic computer operations (M = 4.74)
      - Navigating within the learning management system (M = 4.67)
- **Findings**: 
  - High ratings for importance and ability in course design and communication suggest a need for more faculty support in these areas.
  - Weak positive relationship found between perceived importance and ability to teach online (r(223) = .249, p < .001).
- **Demographic Analysis**:
  - Factors analyzed: Gender, academic rank, online teaching methods, and level taught.
  - Significant differences found in gender and level taught regarding importance of competencies.
  - No significant differences in gender, academic rank, and online teaching methods regarding ability to teach online.
- **Conclusion**: Faculty perceptions of importance align with their perceived abilities, particularly in time management.

# using a simple class signature

In [16]:
from promptsmith.tasks.bad_to_good.bulletize_text_1 import BulletizeText as BT1

bulletize_text_3 = dspy.Predict(BT1)
result_3 = bulletize_text_3(input_text=text_to_bulletize)
bulletized_text_3 = result_3.output_text
print(bulletized_text_3)

- **Research Question 1**: Relationship between faculty's perceptions and ability to teach online
  - Descriptive statistics reported for:
    - Course design
    - Course communication
    - Time management
    - Technical competency
  - High ratings for both constructs (Importance and Ability)
  
- **Course Design Competency**:
  - Highest rated items:
    - Manage grades online (M = 4.73)
    - Creating online assignments (M = 4.68)

- **Course Communication Competency**:
  - Highest rated items:
    - Responding to student questions promptly (M = 4.79)
    - Providing feedback on assignments (M = 4.65)

- **Time Management Competency**:
  - Highest rated items:
    - Scheduling time to design the course prior to delivery (M = 4.65)
    - Spending weekly hours to grade assignments (M = 4.54)

- **Technical Competency**:
  - Highest rated items:
    - Complete basic computer operations (M = 4.74)
    - Navigating within the course in the learning management system (M = 4.67)

- **Fin

# `bulletized_text_3`

- **Research Question 1**: Relationship between faculty's perceptions and ability to teach online
  - Descriptive statistics reported for:
    - Course design
    - Course communication
    - Time management
    - Technical competency
  - High ratings for both constructs (Importance and Ability)
  
- **Course Design Competency**:
  - Highest rated items:
    - Manage grades online (M = 4.73)
    - Creating online assignments (M = 4.68)

- **Course Communication Competency**:
  - Highest rated items:
    - Responding to student questions promptly (M = 4.79)
    - Providing feedback on assignments (M = 4.65)

- **Time Management Competency**:
  - Highest rated items:
    - Scheduling time to design the course prior to delivery (M = 4.65)
    - Spending weekly hours to grade assignments (M = 4.54)

- **Technical Competency**:
  - Highest rated items:
    - Complete basic computer operations (M = 4.74)
    - Navigating within the course in the learning management system (M = 4.67)

- **Findings**:
  - Competencies in course design and communication scored highest
  - Indication of need for more support in these areas

- **Additional Ratings**:
  - Course Design:
    - Manage grades online (M = 4.61)
    - Create online assignments (M = 4.60)
  - Course Communication:
    - Sending announcements/email reminders (M = 4.82)
    - Using email to communicate with learners (M = 4.79)
  - Time Management:
    - Schedule weekly hours to facilitate online course (M = 4.39)
    - Spend weekly hours to grade assignments (M = 4.37)
  - Technical Competency:
    - Complete basic computer operations (M = 4.64)
    - Navigate within the course in the learning management system (M = 4.36)

- **Perception Analysis**:
  - Top competencies of perceived Importance align with perceived Ability
  - Weak positive relationship between Importance and Ability: r(223) = .249, p < .001

- **Research Question 2**: Demographic factors related to perceptions of teaching competencies
  - Demographics analyzed:
    - Gender
    - Academic rank
    - Online teaching methods
    - Level taught
  - Significant differences found:
    - Gender and level taught on importance of competencies
    - Level taught on perceptions of ability to teach online
  - No significant differences in:
    - Gender
    - Academic rank
    - Online teaching methods

- **Conclusion**: Detailed analysis of demographic information collected presented in Table 5, Chapter IV.

# same class signature, but using 'chain of thought'

In [ ]:
bulletize_text_4 = dspy.ChainOfThought(BT1)
result_4 = bulletize_text_4(input_text=text_to_bulletize)
bulletized_text_4 = result_4.output_text
print(bulletized_text_4)

- **Study Question 1**: Relationship between faculty perceptions of:
  - Importance of course design
  - Communication
  - Time management
  - Technical competency
  - Ability to teach online
- **Descriptive Statistics**:
  - Reported in Table 2, page 23
  - High ratings for both constructs: Importance and Ability
- **Course Design Competency**:
  - Highest rated items:
    - Manage grades online (M = 4.73)
    - Creating online assignments (M = 4.68)
- **Course Communication Competency**:
  - Highest rated items:
    - Responding to student questions promptly (M = 4.79)
    - Providing feedback on assignments (M = 4.65)
- **Time Management Competency**:
  - Highest rated items:
    - Scheduling time to design the course prior to delivery (M = 4.65)
    - Spending weekly hours to grade assignments (M = 4.54)
- **Technical Competency**:
  - Highest rated items:
    - Complete basic computer operations (M = 4.74)
    - Navigating within the course in the learning management system (M = 4

# `bulletized_text_4`

- **Study Question 1**: Relationship between faculty perceptions of:
  - Importance of course design
  - Communication
  - Time management
  - Technical competency
  - Ability to teach online
- **Descriptive Statistics**:
  - Reported in Table 2, page 23
  - High ratings for both constructs: Importance and Ability
- **Course Design Competency**:
  - Highest rated items:
    - Manage grades online (M = 4.73)
    - Creating online assignments (M = 4.68)
- **Course Communication Competency**:
  - Highest rated items:
    - Responding to student questions promptly (M = 4.79)
    - Providing feedback on assignments (M = 4.65)
- **Time Management Competency**:
  - Highest rated items:
    - Scheduling time to design the course prior to delivery (M = 4.65)
    - Spending weekly hours to grade assignments (M = 4.54)
- **Technical Competency**:
  - Highest rated items:
    - Complete basic computer operations (M = 4.74)
    - Navigating within the course in the learning management system (M = 4.67)
- **Findings**:
  - High importance scores in course design and communication
  - Need for more support in these competencies
- **Additional Ratings**:
  - Course Design:
    - Manage grades online (M = 4.61)
    - Create online assignments (M = 4.60)
  - Course Communication:
    - Sending announcements/email reminders (M = 4.82)
    - Using email to communicate (M = 4.79)
  - Time Management:
    - Schedule weekly hours to facilitate online course (M = 4.39)
    - Spend weekly hours to grade assignments (M = 4.37)
  - Technical Competency:
    - Complete basic computer operations (M = 4.64)
    - Navigate within the course (M = 4.36)
- **Statistical Analysis**:
  - Weak positive relationship between Importance and Ability: r(223) = .249, p < .001
- **Study Question 2**: Demographic factors related to faculty perceptions:
  - Analyzed demographic information: gender, academic rank, online teaching methods, level taught
  - Results presented in Table 5, page 32, Chapter IV
- **ANOVA Results**:
  - Significant differences in:
    - Gender and level taught on importance of online teaching competencies
  - No significant differences in:
    - Gender, academic rank, online teaching methods
    - Level taught based on perceptions of ability to teach online
- **Conclusion**: No significant differences found in other demographic factors.

# bulletize using chatgpt-4o



## Research Questions & Key Findings

### Research Question 1:
**What is the strength and direction of the relationship between faculty's perceptions of the importance of competencies and their ability to teach online?**

- Descriptive statistics (mean and standard deviation) were reported for:
  - Course Design
  - Course Communication
  - Time Management
  - Technical Competency

- **Highest-rated items (Importance & Ability):**
  - **Course Design:**
    - Manage grades online: M = 4.73 (Importance), M = 4.61 (Ability)
    - Create online assignments: M = 4.68 (Importance), M = 4.60 (Ability)
  - **Course Communication:**
    - Responding to student questions promptly: M = 4.79
    - Providing feedback on assignments: M = 4.65
    - Sending announcements/email reminders: M = 4.82 (Ability)
    - Using email to communicate: M = 4.79 (Ability)
  - **Time Management:**
    - Schedule course design time: M = 4.65
    - Spend weekly hours grading: M = 4.54 (Importance), M = 4.37 (Ability)
    - Schedule weekly hours to facilitate: M = 4.39 (Ability)
  - **Technical Competency:**
    - Complete basic computer operations: M = 4.74 (Importance), M = 4.64 (Ability)
    - Navigate LMS: M = 4.67 (Importance), M = 4.36 (Ability)

- **Conclusion:**
  - Participants rated top competencies similarly for both Importance and Ability.
  - Course design and communication were rated highest in perceived importance.
  - A **weak positive correlation** was found between Importance and Ability:
    - r(223) = 0.249, p < .001

---

### Research Question 2:
**What demographic factors relate to faculty's perceptions of online teaching competencies?**

- Demographics analyzed:
  - Gender
  - Academic rank
  - Online teaching methods
  - Level taught

- **ANOVA Results:**
  - Significant differences found in:
    - **Gender** and **Level Taught** on **Importance**
    - **Level Taught** on **Ability**
  - No significant differences found in:
    - Gender, academic rank, and online teaching methods (on Ability)
    - Academic rank and online teaching methods (on Importance)

## Comparing the five bulletization attempts

| Version | What it looks like | 👍 Pros | 👎 Cons |
|---------|-------------------|--------|---------|
| **`bulletized_text_1`**<br>(plain “text → bullets” function) | A single flat list with no headings. | • Fast<br>• Easy to code | • No structure, so hard to skim.<br>• Numbers and labels mix together. |
| **`bulletized_text_2`**<br>(same function, but with chain-of-thought) | Similar list, marginally tidier ordering. | • Slightly clearer order of points | • Still no real grouping. |
| **`bulletized_text_3`**<br>(class-style signature) | Bullets grouped a bit by topic, but headings are missing. | • Groups related bullets together | • Groups are not named, so you must guess what each set is about. |
| **`bulletized_text_4`**<br>(class + chain-of-thought) | Same as 3. | • Shows why each bullet was chosen | • Still headline-free. |
| **GPT-4o version** | Full mini-report: a main title, clear section titles, one-line section intro, tidy sub-lists, **and a brief conclusion.** | • Easy to scan—titles and bold text guide the eye.<br>• Intro lines set context for each section.<br>• Conclusion wraps things up.<br>• Important numbers stand out.<br>• No stray reasoning text.<br>• Looks ready for a report. | • Slightly longer than the others (adds structure). |

### My pick: GPT-4o

I prefer the **GPT-4o** version because:

* It gives the whole piece a **main title**, so the reader knows the topic right away.  
* Each section starts with a **one-sentence intro** that frames the bullets.  
* Headings and bold words make key facts pop.  
* It ends with a **short conclusion** that ties everything together.  

In short, GPT-4o turns raw notes into a clean, ready-to-publish summary.
